charts/sophora-indexing-service/templates/deployment.yaml

```diff
@@ -5,7 +5,7 @@ metadata:
   labels:
     {{- include "sophora-indexing-service.labels" . | nindent 4 }}
 spec:
-  {{- if not .Values.autoscaling.enabled }}
+  {{- if and (not .Values.autoscaling.enabled) .Values.replicaCount }}
```

Comment:
It's pretty annoying to work with zero values in helm templates, but `0` should be a valid value. 

My suggestion is: 
```
  {{- if and (not .Values.autoscaling.enabled) (or (kindIs "int64" .Values.replicaCount) (kindIs "float64" .Values.replicaCount)) }}
```